### PR TITLE
Add cloudwatch emitter

### PR DESCRIPTION
**Pull request by ish-bindra**
This PR
- Adds a cloudwatch emitter package that sends out the following metrics:
  - **Completed** 
  - **Stuck** - Jobs that are stuck and are too risky to restart
  - **Restarted** - Jobs that got stuck but restarted
  - **Pending** - Pending jobs that should have been run but not running due to less capacity
 This emitter hooks into event emitter in the job scheduler libraries
- Adds a initialisation check that verifies the maintenance job's existence
- Adds a new `pending` event that emits job name against pending jobs count

## Usage:
```typescript
const jobScheduler = new JobScheduler({
       ...
});

schedulerMetrics(jobScheduler, 'ordermentum');

export default jobScheduler;
```

Screenshot 2023-12-22 at 9 02 26 am
Screenshot 2023-12-22 at 9 02 56 am

### CODE REVIEW

File: packages/cloudwatch/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+const client = new CloudWatchClient();
+const Namespace = 'OM-DB-Jobs';
+
+export const schedulerMetrics = (
+  scheduler: PrismaScheduler | SequelizeScheduler,
+  service: string
+) => {
+  scheduler.events.on('duration', (job: Job) => {
+    const MetricName = job!.name.toUpperCase();
```
Comment (johndagostino) [inline packages/cloudwatch/src/index.ts:29]: i kind of dislike the bang operator, i think its a code smell

why is name optional?

File: packages/cloudwatch/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+import {
+  JobAttributes as SequelizeJob,
+  JobInstance,
+} from '@steveojs/scheduler-sequelize/lib/models/job';
+
+type Job = PrismaJobContext['job'] | JobInstance;
+type JobAttributes = SequelizeJob | PrismaJobContext['job'];
+
+const client = new CloudWatchClient();
+const Namespace = 'OM-DB-Jobs';
```
Comment (rrsmith) [inline packages/cloudwatch/src/index.ts:22]: (trivial) should make this 'steveo' rather than 'OM'

File: packages/cloudwatch/src/index.ts (hunk 1 above)
Comment (ish-bindra) [reply packages/cloudwatch/src/index.ts]: I originally intended to make no changes in the schedulers themselves. 
The job type from the prisma scheduler was not exported so had to do a workaround.

I've exported it now and removed the bangs.

**Review by rrsmith — APPROVED** [review]

**Comment by ish-bindra** [comment]
@johndagostino Good to merge?

**Review by johndagostino — APPROVED** [review]

**Pull request merged**